### PR TITLE
Add support for STORAGE_EMULATOR_HOST like the offical GCS go sdk.

### LINKED ISSUE #4: Add support for STORAGE_EMULATOR_HOST in gmutex
I'd love to be able to test my code using [fake-gcs-server](https://github.com/fsouza/fake-gcs-server/tree/main), but currently there's no way to override the [hostname](https://github.com/ncruces/go-gcp/blob/master/gmutex/gmutex.go#L508) for GCS.

The [official GCS golang library](https://pkg.go.dev/cloud.google.com/go/storage#hdr-Creating_a_Client) uses the environment variable `STORAGE_EMULATOR_HOST` to allow tests to easily inject a different hostname into GCS client code. Perhaps we could follow a similar pattern in this library?

Would you be willing to review and possibly accept a PR from adding support for STORAGE_EMULATOR_HOST?

### PR DESCRIPTION
**Pull request by brycefisher**
Closes https://github.com/ncruces/go-gcp/issues/4

### CODE REVIEW

**Comment by ncruces** [comment]
I'll probably move some of the documentation to the README, and/or package level documentation, after testing with pkgsite. Hope you don't mind!

**Comment by brycefisher** [comment]
Have at it!

I noticed this PR doesn't work with `fake-gcs-server` because that project only implements some of the JSON based endpoints for GCS (`/storage/v1/b/<bucket>/o/<object>`), whereas this project relies on the XML endpoints (`/<bucket>/<object>`). So, I haven't been able to get the tests in this project to pass using `fake-gcs-server`. 

I'm currently investigating `github.com/fullstorydev/emulators/storage/gcsemu`, but it appears that project doesn't support the `PUT` method (although it does recognize the XML urls used by go-gcp): https://github.com/fullstorydev/emulators/blob/master/storage/gcsemu/gcsemu.go#L123

**Comment by ncruces** [comment]
That's unfortunate…

I used the XML API because it's much simpler for the use case, but what would really worry me is whether test servers respect the [consistency](https://cloud.google.com/storage/docs/consistency) expectations that underpin the project.

**Comment by brycefisher** [comment]
The emulators project uses locks to a backing in memory / filesystem store that seem possibly sufficient for those consistency guarantees.

I've forked the emulator project and have two of the three tests passing. Seems like I'm having trouble getting the x-good-meta-TTL headers back into the response from Head requests. If I can it to a good place, I'll see if those maintainers would accept a patch for additional endpoints.

It's wild that GCP has so many emulators but none for GCS, lol.

**Comment by brycefisher** [comment]
Working locally, I have a few changes to gmutex TestMain for toggling to http and insecure TLS.  I'll try to push those up soon for review as well

**Comment by brycefisher** [comment]
Ok! There's a non-hacky way to control where gmutex will try to connect to the server -- host, scheme and insecure all controlled through 3 separate envvars, defaults to real GCS.

I've managed to fix the test of lock expiration (for the Abandon() use case) in [my fork](https://github.com/fullstorydev/emulators/compare/master...brycefisher:emulators:feat/gcsemu-put-method?expand=1) of `github.com/fullstorydev/emulators`, but broke the contention test again 😂 I'll see if I can figure this out Monday -- I'd love to get the endpoints needed for gmutex to be capable of running "offline" in CI and in parallel with dynamically chosen ports. Feels like I'm very close!!

I think technically, aside from whatever feedback you have here, the changes to go-gcp are done. Many thanks for being so receptive to contributions!! 🎊

**Comment by ncruces** [comment]
Apparently, to be compatible with `STORAGE_EMULATOR_HOST` we should use `STORAGE_EMULATOR_HOST=http://localhost:9023`? Idk, I've never used this.

Do you have any reference for `STORAGE_EMULATOR_SCHEME`?

**Comment by brycefisher** [comment]
Good catch!! I misunderstood. I'll update the PR with that and remove scheme environment variable

**Comment by ncruces** [comment]
I feel inclined to set the `InsecureSkipVerify` if the same variable is set, unless you feel strongly against it.
The less knobs the better.

File: gmutex/gmutex.go
Comment (brycefisher) [inline gmutex/gmutex.go:529]: I didn't add support for incorporating a path prefix from the environment variable, but I assume that's ok? Happy to change if you like!

File: gmutex/gmutex.go
Comment (brycefisher) [inline gmutex/gmutex.go:58]: Would you like me to remove additions to the godocs?

**Comment by ncruces** [comment]
I'm sorry for being delayed reviewing this. Work got in the way.

I'd like to do a proper review, then do a release.
Are you blocked?

**Comment by brycefisher** [comment]
You're delightful and I'm not blocked by this PR. Take your time!

**Review by ncruces — APPROVED** [review]
Sorry this slipped through.
Thanks!

File: gmutex/gmutex.go
Comment (ncruces) [reply gmutex/gmutex.go:58]: Tweaked in a follow up.
Thanks though!

**Pull request merged**
